src/test_util.rs

```diff
@@ -4,25 +4,24 @@
 
 
 use std::sync::mpsc::{channel, Sender};
-use std::thread;
 use glib::*;
 
 #[allow(dead_code)]
 pub fn run_async<T: Send + 'static, Q: FnOnce(Sender<T>, MainLoop) + Send + 'static>(start: Q) -> T {
-    let l = MainLoop::new(None, false);
-    let c = MainContext::default().unwrap();
+    let c = MainContext::new();
+    let l = MainLoop::new(&c, false);
     let l_clone = l.clone();
 
     let (tx, rx) = channel();
 
-    thread::spawn(move || {
-        c.invoke(move || {
-            start(tx, l_clone);
-        });
+    c.invoke(move || {
+        let c = l_clone.get_context().unwrap();
+        c.push_thread_default();
```

Comment:
I would do that simply at the top/bottom of the run_async function to have it for the whole runtime of the test

Reply:
Updated accordingly.